src/curve/stable/StablePair.sol

```diff
@@ -159,24 +161,24 @@ contract StablePair is ReservoirPair {
         // and use the current totalSupply of LP tokens for calculations since there is no new
         // LP tokens minted for platformFee
         uint256 _totalSupply;
-        try StablePair(this).mintFee(balance0, balance1) returns (uint256 rTotalSupply, uint256) {
+        try StablePair(this).mintFee(_reserve0, _reserve1) returns (uint256 rTotalSupply, uint256) {
             _totalSupply = rTotalSupply;
         }
         catch {
             _totalSupply = totalSupply;
         }
 
-        amount0 = (liquidity * balance0) / _totalSupply;
-        amount1 = (liquidity * balance1) / _totalSupply;
+        amount0 = (liquidity * _reserve0) / _totalSupply;
+        amount1 = (liquidity * _reserve1) / _totalSupply;
 
         _burn(address(this), liquidity);
         _safeTransfer(token0, to, amount0);
         _safeTransfer(token1, to, amount1);
 
         _update(_totalToken0(), _totalToken1(), reserve0, reserve1);
 
-        lastLiquidityEventReserve0 = reserve0;
-        lastLiquidityEventReserve1 = reserve1;
+        lastInvariant = uint192(_computeLiquidity(reserve0, reserve1));
+        lastInvariantAmp = _getCurrentAPrecise();
```

Comment:
Can gas be saved using `getReserves` to batch fetch - or does solidity already optimize this under the hood?

Reply:
I tried doing 

```
(_reserve0, _reserve1, ) = getReserves();
lastInvariant = uint192(_computeLiquidity(_reserve0, _reserve1));
```

somehow it actually costs (a little) more gas. Note that the variables are not new, they're re-used. When I use new variables the results are the same

Reply:
therefore, resolving for now